include/arch/arm64/barrier.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022 Alexandre Mergnat <amergnat@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_ARM64_BARRIER_H_
+#define ZEPHYR_INCLUDE_ARCH_ARM64_BARRIER_H_
+
+#ifndef _ASMLANGUAGE
+
+/*
+ * Provide a default implementation for un-supported compilers.
+ */
```

Comment:
Please add another commit on top to get rid of these defines in `include/arch/arm64/lib_helpers.h`

Reply:
ACK